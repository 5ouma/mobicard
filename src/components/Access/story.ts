import type { Props } from "./Access.astro";

export const Default: Props = {
  homepage: "https://example.com",
  email: "email@example.com",
};

export const SubDomain: Props = {
  homepage: "https://www.example.com",
  email: "email@www.example.com",
};

export const WithoutEmail: Props = {
  homepage: "https://www.example.com",
};
