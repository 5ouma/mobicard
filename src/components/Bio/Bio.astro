---
import { Image } from "astro:assets";

import "../../styles/global.css";

export type Props = {
  name: string;
  description: string;
  icon: string;
};
const props = Astro.props;
---

<div id="bio">
  <div id="profile">
    <Image src={props.icon} alt={`${props.name}'s avatar`} inferSize={true} />
    <h1>{props.name}</h1>
  </div>
  <p>{props.description}</p>
</div>

<style>
  #bio {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    gap: 0.5rem;
  }

  #profile {
    display: flex;
    align-items: center;
    gap: 1rem;
  }

  img {
    width: 6rem;
    height: auto;
  }

  h1,
  p {
    margin: 0;
    font-family: "LINE Seed JP", sans-serif;
    font-style: normal;
    line-height: normal;
    color: var(--foreground);
  }

  h1 {
    font-size: 2.5rem;
    font-weight: bold;
  }

  p {
    font-size: 0.9rem;
    font-weight: normal;
    white-space: pre-line;
  }
</style>
