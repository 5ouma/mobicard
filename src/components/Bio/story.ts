import type { Props } from "./Bio.astro";

export const Default: Props = {
  name: "Bio Name",
  description: "Bio Description",
  icon: "https://github.com/withastro.png",
};

export const Japanese: Props = {
  name: "名前",
  description: "説明",
  icon: "https://github.com/withastro.png",
};

export const Long: Props = {
  name: "Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Name",
  description:
    "Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Description",
  icon: "https://github.com/withastro.png",
};
