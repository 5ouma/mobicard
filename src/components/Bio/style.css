#bio {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  gap: 0.3rem;
}

#profile {
  display: flex;
  align-items: center;
  gap: 0.3rem;
}

img {
  width: 6rem;
  height: auto;
}

span {
  font-family: "LINE Seed JP" sans-serif;
  font-style: normal;
  line-height: normal;
  color: var(--foreground);
}

#name {
  font-size: 2.5rem;
  font-weight: bold;
}

#description {
  font-size: 0.9rem;
  font-weight: normal;
  white-space: pre-line;
}
