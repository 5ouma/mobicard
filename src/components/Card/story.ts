import type { Props } from "./Card.astro";

export const Default: Props = { url: "https://example.com" };
export const SubDomain: Props = { url: "https://www.example.com" };
