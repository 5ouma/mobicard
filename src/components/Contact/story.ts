import type { Props } from "./Contact.astro";

export const Bluesky: Props = {
  service: "Bluesky",
  id: "bsky.app",
};

export const Facebook: Props = {
  service: "Facebook",
  id: "facebook",
};

export const GitHub: Props = {
  service: "GitHub",
  id: "github",
};

export const Instagram: Props = {
  service: "Instagram",
  id: "instagram",
};

export const Mastodon: Props = {
  service: "Mastodon",
  id: "@Mastodon@mastodon.social",
};

export const Misskey: Props = {
  service: "Misskey",
  id: "@support@misskey.io",
};

export const Posts: Props = {
  service: "Posts",
  id: "support",
};

export const Twitter: Props = {
  service: "Twitter",
  id: "twitter",
};

export const Mail: Props = {
  service: "Mail",
  id: "mail@example.com",
};
