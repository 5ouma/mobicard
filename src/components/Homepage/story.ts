import type { Props } from "./Homepage.astro";

export const Domain: Props = { url: "https://example.com" };
export const Path: Props = { url: "https://www.example.com/path" };
export const Japanese: Props = { url: "https://www.example.com/パス" };
