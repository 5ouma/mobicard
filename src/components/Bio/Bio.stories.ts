import Bio from "./Bio.astro";
import * as stories from "./story.ts";

export default {
  title: "Bio",
  component: Bio,
};

export const Default = { args: stories.Default };
export const Japanese = { args: stories.Japanese };
export const Long = { args: stories.Long };
