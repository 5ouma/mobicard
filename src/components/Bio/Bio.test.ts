import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { describe, expect, test } from "vitest";

import Bio from "./Bio.astro";
import * as stories from "./story.ts";

describe("Bio", () => {
  for (const [name, props] of Object.entries(stories)) {
    test(name, async () => {
      const container: AstroContainer = await AstroContainer.create();
      const result: string = await container.renderToString(Bio, { props });

      expect(result).toContain(`>${props.name}</span>`);
      expect(result).toContain(`>${props.description}</span>`);
      expect(result).toContain(`<img src="${props.icon}"`);
    });
  }
});
