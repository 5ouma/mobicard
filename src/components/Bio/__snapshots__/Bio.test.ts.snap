// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bio > Default 1`] = `"<div id="bio" data-astro-cid-yljsttrw> <div id="profile" data-astro-cid-yljsttrw> <picture data-astro-cid-yljsttrw> <source srcset="https://github.com/withastro.png" type="image/avif"><source srcset="https://github.com/withastro.png" type="image/webp"><source srcset="https://github.com/withastro.png" type="image/jpeg"> <img src="https://github.com/withastro.png" data-image-component="true" alt="Bio Name's avatar" data-astro-cid-yljsttrw width="460" height="460" loading="lazy" decoding="async"> </picture> <h1 data-astro-cid-yljsttrw>Bio Name</h1> </div> <p data-astro-cid-yljsttrw>Bio Description</p> </div> "`;

exports[`Bio > Japanese 1`] = `"<div id="bio" data-astro-cid-yljsttrw> <div id="profile" data-astro-cid-yljsttrw> <picture data-astro-cid-yljsttrw> <source srcset="https://github.com/withastro.png" type="image/avif"><source srcset="https://github.com/withastro.png" type="image/webp"><source srcset="https://github.com/withastro.png" type="image/jpeg"> <img src="https://github.com/withastro.png" data-image-component="true" alt="名前's avatar" data-astro-cid-yljsttrw width="460" height="460" loading="lazy" decoding="async"> </picture> <h1 data-astro-cid-yljsttrw>名前</h1> </div> <p data-astro-cid-yljsttrw>説明</p> </div> "`;

exports[`Bio > Long 1`] = `"<div id="bio" data-astro-cid-yljsttrw> <div id="profile" data-astro-cid-yljsttrw> <picture data-astro-cid-yljsttrw> <source srcset="https://github.com/withastro.png" type="image/avif"><source srcset="https://github.com/withastro.png" type="image/webp"><source srcset="https://github.com/withastro.png" type="image/jpeg"> <img src="https://github.com/withastro.png" data-image-component="true" alt="Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Name's avatar" data-astro-cid-yljsttrw width="460" height="460" loading="lazy" decoding="async"> </picture> <h1 data-astro-cid-yljsttrw>Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Name</h1> </div> <p data-astro-cid-yljsttrw>Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Description</p> </div> "`;
